### PR TITLE
Modify MNIST async example to store test dataset on separate worker and evaluate remotely

### PR DESCRIPTION
**Pull request by midokura-silvia**
Models are sent and evaluated at the (remote) worker.

### CODE REVIEW

File: examples/tutorials/advanced/websockets-example-MNIST-parallel/run_websocket_client.py [hunk 1]
```diff
@@ -107,59 +107,54 @@ def define_and_get_arguments(args=sys.argv[1:]):
...
-    np.set_printoptions(formatter={"float": "{: .0f}".format})
-    for worker_id, worker_model, _ in results:
-        evaluate_model(worker_id, worker_model, "cpu", test_loader, print_target_hist=False)
+def evaluate_model_on_worker(
+    model_identifier, worker, dataset_key, model, nr_bins, batch_size, print_target_hist=False
+):
+    model.eval()
 
+    # Create and send train config
+    train_config = sy.TrainConfig(
```
[the remark this answers is not in the document]
Comment (midokura-silvia) [reply examples/tutorials/advanced/websockets-example-MNIST-parallel/run_websocket_client.py]: yes. otherwise we need to share: model, potentially loss function and batch size.

File: examples/tutorials/advanced/websockets-example-MNIST-parallel/run_websocket_server.py [hunk 2]
```diff
@@ -67,14 +72,27 @@ def start_websocket_server_worker(id, host, port, hook, verbose, keep_labels=Non
     data, target = utils.create_gaussian_mixture_toy_data(nr_samples=100)
     server.add_dataset(sy.BaseDataset(data, target), key="gaussian_mixture")
 
+    # Setup partial iris dataset
```
[the remark this answers is not in the document]
Comment (midokura-silvia) [reply examples/tutorials/advanced/websockets-example-MNIST-parallel/run_websocket_server.py]: The iris dataset is not used in the tutorial at all. Currently it is added out of pure convenience for testing. There is still an issue with the unit tests. Sometimes they fail when they start a WebsocketServerWorker. And then it is handy to use an independent server that has all the datasets.

[the remark this answers is not in the document]
Comment (mari-linhares) [reply examples/tutorials/advanced/websockets-example-MNIST-parallel/run_websocket_server.py]: Hmm, I see... maybe we can move this to a helper script in the test folder?

File: test/torch/hook/test_hook_args.py [hunk 3]
```diff
@@ -0,0 +1,18 @@
+from syft.frameworks.torch.hook import hook_args
```
[the remark this answers is not in the document]
Comment (midokura-silvia) [reply test/torch/hook/test_hook_args.py]: The hook_args module didn't have any unit tests. I needed to debug the behaviour of build_rule for numpy values so I added a first test.

File: syft/federated/federated_client.py [hunk 4]
```diff
@@ -122,3 +123,66 @@ def _fit(self, model, dataset_key, loss_fn):
...
+                    hist, _ = np.histogram(target, bins=nr_bins, range=(0, nr_bins))
+                    hist_target += hist
+                    hist, _ = np.histogram(pred, bins=nr_bins, range=(0, nr_bins))
+                    hist_pred += hist
+                correct += pred.eq(target.view_as(pred)).sum().item()
+
+        if calculate_loss:
+            test_loss /= len(data_loader.dataset)
+
+        return test_loss, correct, len(data_loader.dataset), hist_pred, hist_target
```
[the remark this answers is not in the document]
Comment (midokura-silvia) [reply syft/federated/federated_client.py]: I changed the return value into a dictionary. That way we keep the return signature constant.
